src/main/java/com/avispl/symphony/dal/communicator/ppdswave/PhilipsWaveAggregatorCommunicator.java

```diff
@@ -0,0 +1,1019 @@
+    }
+
+    /**
+     * Volume change command
+     *
+     * @param displayId to execute command for
+     * @param volumeLevel new volume level
+     */
+    private void commandChangeVolume(String displayId, String volumeLevel) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.VOLUME, displayId, Float.parseFloat(volumeLevel)), JsonNode.class);
+    }
+
+    /**
+     * Screen brightness change
+     *
+     * @param displayId to execute command for
+     * @param brightnessLevel new brightness level
+     */
+    private void commandChangeBrightness(String displayId, String brightnessLevel) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.BRIGHTNESS, displayId, Float.parseFloat(brightnessLevel)), JsonNode.class);
+    }
+
+    /**
+     * Power state change command, ON/STANDBY
+     *
+     * @param displayId to execute command for
+     * @param powerState new power state
+     */
+    private void commandChangePowerState(String displayId, String powerState) throws Exception {
+        String powerStateValue = "1".equals(powerState) ? "ON" : "STANDBY";
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.POWER, displayId, powerStateValue), JsonNode.class);
+    }
+
+    /**
+     * Screen orientation change, LANDSCAPE/PORTRAIT
+     *
+     * @param displayId to execute command for
+     * @param orientationState new orientation state
+     */
+    private void commandChangeOrientation(String displayId, String orientationState) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.ORIENTATION, displayId, orientationState), JsonNode.class);
+    }
+
+    /**
+     * Video input source change
+     *
+     * @param displayId to execute command for
+     * @param inputState new input state
+     */
+    private void commandChangeInput(String displayId, String inputState) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.INPUT, displayId, inputState), JsonNode.class);
+    }
+
+    /**
+     * Take screenshot command
+     *
+     * @param displayId to execute command for
+     */
+    private void commandTakeScreenshot(String displayId) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.SCREENSHOT, displayId), JsonNode.class);
+    }
+
+    /**
+     * Change IR control mode command, LOCKED/POWER_ONLY/UNLOCKED/VOLUME_ONLY
+     *
+     * @param displayId to execute command for
+     * @param irMode new IR control mode
+     */
+    private void commandChangeIRMode(String displayId, String irMode) throws Exception {
+        doPost("/graphql", String.format(Constants.GraphQLRequests.ControlRequest.IR_MODE, displayId, irMode), JsonNode.class);
```

Comment:
Should we use try/catch for each control function and handle a message for each control command when an error occurs? the same for commandChangeKeyboardMode, commandChangeIRMode,commandTakeScreenshot,commandChangeInput,... etc

Reply:
Well, any error thrown will be propagated to the top, so there's no real reason to handle it in any specific manner, unless we'll end up having a general error message format shown in popups on UI or something like that, which we dont have at this point, unfortunately (as well as the custom popups are not yet supported there). 

So basically, once there's an exception during the command execution - it will be shown in a popup on UI, thus guaranteeing visibility over the command succession.

Reply:
I'm just concerned that when the user controls a particular property, it will display a notification that may leave the user unsure of what it means at that moment. So, I think we need to handle it with some general message or, as per your suggestion, we can keep it as it is. That's just my personal opinion, haha :)

Reply:
Oh, so you mean to rethrow the exception, but with the meaningful type and message, to keep it consistent? 
I mean, we need to propagate it to the UI anyways...